File: backend/src/main/java/ProgramaClip/ClipMoneyAPI/entidad/PlazoFijo.java
package ProgramaClip.ClipMoneyAPI.entidad;

import java.sql.Date;
import java.sql.Time;
import java.sql.Timestamp;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

import com.sun.istack.NotNull;

import lombok.Data;

@Entity
@Data
public class PlazoFijo {
	@Id
	@GeneratedValue
	private Long id;
	@NotNull
	private Timestamp fechaHoraCreacion;
	@NotNull
	private Date fechaInicioPlazo;
	@NotNull
	private Date fechaFinPlazo;
	@NotNull
	private Float interesAnual;
	@NotNull
	private Float monto;
	@ManyToOne
	@NotNull
	private Cuenta cuenta;
	@ManyToOne
	@NotNull
	private TipoPlazoFijo tipoPlazoFijo;
}

File: Models/PlazoFijo.cs
namespace proyecto_clip_money_e_proyecto_clip.Models
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Data.Entity.Spatial;

    [Table("PlazosFijos")]
    public partial class PlazoFijo
    {
        public long Id { get; set; }

        [Column(TypeName = "timestamp")]
        [MaxLength(8)]
        [Timestamp]
        public byte[] FechaHoraCreacion { get; set; }

        [Column(TypeName = "date")]
        public DateTime FechaInicioPlazo { get; set; }

        [Column(TypeName = "date")]
        public DateTime FechaFinPlazo { get; set; }

        public double InteresAnual { get; set; }

        public double Monto { get; set; }

        public long CuentaNroCuenta { get; set; }

        public int TipoPlazoFijoId { get; set; }

        public virtual Cuenta Cuenta { get; set; }

        public virtual TipoPlazoFijo TipoPlazoFijo { get; set; }
    }
}
